Extend the Log2Influx example to stress test the grafana dashboard …

khpandya opened the pull request:
…by logging in 15 values every second through influxdb.

This commit contains just one python file which basically creates random values for measurements such as pressure in millibar or temperature in Celsius. About 15 measurements are randomly generated every second and logged into an Influx database. This is intended to be used to stress test the Grafana dashboard.

khpandya committed:
Extended the Log2Influx example to stress test the grafana dashboard by logging in 15 values every second through influxdb

khpandya commented:
Apologies if I got something wrong here. 
I've been really busy and didn't have a time to check out a lot of other pull requests before I posted my first one.

khpandya commented:
All suggestions appreciated

Burke-Daniel commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,170 @@
+                "host": "server01",
+                "region": "us-west"
+            },
+            #"time": f"{datetime.now()}",
```
So ideally there will be no commits with code commented out, so if you accidentally made a commit with commented out code you could do an interactive rebase, and basically rewrite history and edit this commit, but alternatively for now you can add another commit to this PR which removes that commented line

colton-smith commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,170 @@
+                "Pascal": 3.0,
+                }
+         },
+        
+        
```
There isn't really a standard for formatting json-like objects, but consistency is good. For example, you have two lines here, but you have 1 line in my above comment.

colton-smith commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,170 @@
+                "Fahrenheit": 3,
+               }
+        },
+        
```
See below comment

colton-smith commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,170 @@
+        print(f"s22: {telemetry[5]['fields']['samplef22']}")
+        print(f"s23: {telemetry[5]['fields']['samplef23']}")
+        print()
+        
+
```
Just a single line here

colton-smith commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,170 @@
+        telemetry[5]['fields']['samplef21'] = random.randint(10,15)
+        telemetry[5]['fields']['samplef22'] = random.randint(30,35)
+        telemetry[5]['fields']['samplef23'] = random.randint(130,135)
+        
+
```
Single line here as well

colton-smith commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,170 @@
+# Example for logging data to influxdb using python
+# Sourced from the influxdb python docs
+
+import random
+import time
+from datetime import datetime 
+from influxdb import InfluxDBClient
```
Imports should be ordered alphabetically, they should be as follows:
first-party imports (random, time, anything default to python)

third-party imports (influx, any other packages)

user-defined imports (Anything you wrote and are importing)

colton-smith reviewed (changes requested):
Looks good, just a few added lines that are redundant and import ordering. Great Job!

khpandya committed:
white-space formatting fixes

Follows @colton-smith's comments on the PR for the previous commit

khpandya reviewed (commented):
Made the changes, would be great if someone could review it again.

Burke-Daniel commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,164 @@
+                "Metre": 1.0,
+                "Feet": 3.0,
+                }
+        },
```
So if you look at the indentation of this bracket compared to the others, you'll see that it appears that they are not indented consistently. It would be nicer if they are all indented the same amount.

Burke-Daniel reviewed (changes requested):
So the program seems to run fine twice, I get the output in the console, and then after that, I am getting this error for some reason:
```
Traceback (most recent call last):
  File "log-windtunnel-influx.py", line 164, in <module>
    main()
  File "log-windtunnel-influx.py", line 127, in main
    client.write_points(telemetry)
  File "C:\Users\Dev\AppData\Local\Programs\Python\Python38\lib\site-packages\influxdb\client.py", line 525, in write_points
    return self._write_points(points=points,
  File "C:\Users\Dev\AppData\Local\Programs\Python\Python38\lib\site-packages\influxdb\client.py", line 594, in _write_points
    self.write(
  File "C:\Users\Dev\AppData\Local\Programs\Python\Python38\lib\site-packages\influxdb\client.py", line 337, in write
    self.request(
  File "C:\Users\Dev\AppData\Local\Programs\Python\Python38\lib\site-packages\influxdb\client.py", line 302, in request
    raise InfluxDBClientError(response.content, response.status_code)
influxdb.exceptions.InfluxDBClientError: 400: {"error":"partial write: field type conflict: input field \"samplef01\" on measurement \"samplem1\" is type integer, already exists as 
type float dropped=3"}
```
With my limited knowledge of this, it seems like the error is talking about something to do with the type of one of the data fields. Would you be able to take a look and see if you can reproduce (it could be something on my end), as well as the other couple of comments I have? Overall, great work on this!

khpandya committed:
Removed a comment

khpandya committed:
fixed bug 

values assigned with randint should be declared as an integer in the telemetry list and the ones  assigned with gauss should be declared as floats

khpandya committed:
formatting fix

khpandya commented:
Changes made, re-review requested

khpandya reviewed (commented):
code runs as expected now

Burke-Daniel commented on ControlLaptop/Examples/Log-windtunnel-influx/log-windtunnel-influx.py:
```diff
@@ -0,0 +1,162 @@
+            "tags": {
+                "host": "server01",
+                "region": "us-west"
+            }
```
Missing a comma here in this dict

khpandya replied:
Done!

Burke-Daniel reviewed (changes requested):
See the one comment I made there, other than that great work on this.

khpandya committed:
Formatting fix

colton-smith reviewed (approved):
GTM

Burke-Daniel reviewed (approved):
Good job, GTM
